custom-elements/custom-elements-registry/define.html

```diff
@@ -0,0 +1,230 @@
+    "annotation-xml",
+    "color-profile",
+    "font-face",
+    "font-face-src",
+    "font-face-uri",
+    "font-face-format",
+    "font-face-name",
+    "missing-glyph",
+  ];
+  validCustomElementNames.forEach(function (name) {
+    test(function () {
+      window.customElements.define(name, class {});
+    }, 'If name is ' + name + ', success');
+  });
+  invalidCustomElementNames.forEach(function (name) {
+    test(function () {
+      assert_throws("SyntaxError", function () {
+        window.customElements.define(name, class {});
+      });
+    }, 'If name is ' + name + ', throw a SyntaxError');
+  });
+
+  // 3. If this CustomElementsRegistry contains an entry with name name,
+  // then throw a NotSupportedError and abort these steps.
+  test(function () {
+    window.customElements.define("test-define-dup-name", class {});
+    assert_throws("NOT_SUPPORTED_ERR", function () {
+      window.customElements.define("test-define-dup-name", class {});
+    });
+  }, "If the name is already defined, throw a NotSupportedError");
+
+  // 4. If this CustomElementsRegistry contains an entry with constructor constructor,
+  // then throw a NotSupportedError and abort these steps.
+  test(function () {
+    class testDupCtor {};
+    window.customElements.define("test-define-dup-ctor", testDupCtor);
+    assert_throws("NOT_SUPPORTED_ERR", function () {
+      window.customElements.define("test-define-dup-ctor2", testDupCtor);
+    });
+  }, "If ctor is already defined, throw a NotSupportedError");
+
+  // 7.1. If extends is a valid custom element name,
+  // then throw a NotSupportedError.
+  validCustomElementNames.forEach(function (name) {
+    test(function () {
+      assert_throws("NOT_SUPPORTED_ERR", function () {
+        window.customElements.define("test-define-extend-valid-name", class {}, { extends: name });
+      });
+    }, 'If extends is ' + name + ', throw a NotSupportedError');
+  });
+
+  // 7.2. If the element interface for extends and the HTML namespace is HTMLUnknownElement
+  // (e.g., if extends does not indicate an element definition in this specification),
+  // then throw a NotSupportedError.
+  [
+    // https://html.spec.whatwg.org/multipage/dom.html#elements-in-the-dom:htmlunknownelement
+    "bgsound",
+    "blink",
+    "isindex",
+    "multicol",
+    "nextid",
+    "spacer",
+    "elementnametobeunknownelement",
+  ].forEach(function (name) {
+    test(function () {
+      assert_throws("NOT_SUPPORTED_ERR", function () {
+        window.customElements.define("test-define-extend-" + name, class {}, { extends: name });
+      });
+    }, 'If extends is ' + name + ', throw a NotSupportedError');
+  });
+
+  // 8. Let observedAttributesIterable be Get(constructor, "observedAttributes").
+  // Rethrow any exceptions.
+  // See step 12 for rethrow tests.
+
+  // 10. Let prototype be Get(constructor, "prototype"). Rethrow any exceptions.
+  function throw_rethrown() {
```

Comment:
Use a function name that describes what this helper does, and not who uses it: This doesn't actually throw something rethrown; it doesn't even rethrow anything. It just throws something.